core/utils/histogram.h

```diff
@@ -64,13 +65,60 @@ class Histogram {
   Histogram(size_t num_buckets, T bucket_width)
       : bucket_width_(bucket_width), count_(), buckets_(num_buckets + 1) {}
 
+  // Swap operator allows clean summarizing with external lock,
+  // when using a histogram on an active data stream.
+  // Create a new, empty histogram of the appropriate size
+  // while NOT holding a lock, then use tmphist.swap(datahist)
+  // while holding a lock.  Release the lock, and summarize
+  // from tmphist while new data accumulate into datahist.
+  void swap(Histogram &other) noexcept {
+    using std::swap;
+    swap(bucket_width_, other.bucket_width_);
+    // atomic count_ field requires manual swap
```

Comment:
While it's not directly related to this PR; perhaps we can remove `count_` from the class entirely? It would reduce code verbosity regarding atomic access of the counter, also eliminating the non-atomicity between the counter and the actual buckets. The variable doesn't seem to serve anything useful (`Summarize()` can be modified to not rely on `count_` at all).

Reply:
I like that idea, so I implemented it.  It simplifies the rest of the code a bunch.  See what you think...